lib/crowi/index.js

```diff
@@ -0,0 +1,218 @@
+'use strict';
+
+
+var debug = require('debug')('crowi:crowi')
+  , pkg = require('../../package.json')
+  , path = require('path')
+  , sep = path.sep
+  , Promise = require('bluebird')
+
+  , http     = require('http')
+  , express  = require('express')
+
+  , mongoose    = require('mongoose')
+
+  , models = require('../models')
+  ;
+
+function Crowi (rootdir, env)
+{
+  var self = this;
+
+  this.version = pkg.version;
+
+  this.rootDir   = rootdir;
+  this.pluginDir = path.join(this.rootDir, 'node_modules') + sep;
+  this.publicDir = path.join(this.rootDir, 'public') + sep;
+  this.libDir    = path.join(this.rootDir, 'lib') + sep;
+  this.viewsDir  = path.join(this.libDir, 'views') + sep;
+  this.mailDir   = path.join(this.viewsDir, 'mail') + sep;
+
+  this.config = {};
+  this.mailer = {};
+
+  this.models = {};
+
+  this.env = env;
+  this.node_env = this.env.NODE_ENV || 'development';
+  this.port = this.env.PORT || 3000;
+
+  if (this.node_env == 'development') {
+    Promise.longStackTraces();
+  }
+
+  //time.tzset('Asia/Tokyo');
+};
+
+Crowi.prototype.init = function() {
```

Comment:
I think the interface of `Crowi.init` below is better because we only depend on the envinronment variables at one place.

```javascript
// lib/crowi/index.js
Crowi.prototype.init = function(config) {
};
```